### ISSUE TITLE
Disallow null if not option-wrapped everywhere, not just for record fields

### ISSUE THREAD

**Issue by cmeeren** [Tarmil/FSharp.SystemTextJson#158]
The document [Customizing the serialization format](https://github.com/Tarmil/FSharp.SystemTextJson/blob/master/docs/Customizing.md) says:

> By default, FSharp.SystemTextJson throws an exception when the following conditions are met:
> 
> * **it is deserializing a record or a union;**
> * a field's JSON value is null or the field is unspecified;
> * that field's type isn't an explicitly nullable F# type (like option, voption and Skippable).

What is the reason for the first requirement (emphasis mine)?

I ask because I was recently bitten by a bug in production. I am deserializing a JSON document to the type `MyRootRecord`, but when the JSON body itself is `null`, then the returned `MyRootRecord` is also null (even though it is a record and F# does not allow null values of it), causing some `NullReferenceExecption` elsewhere in the code.

I would expect that if I deserialize a top-level `null` value to a record/union (not wrapped in `option` or similar), then I get a deserialization exception similar to if it had been a record field.

In other words, the "throw on null" behavior should be governed by the target type, not by whether or not it's a property. (This could therefore also apply to array items, which I haven't tested.)

**Comment by Tarmil** [comment]
Good point, when the value itself is a record or union, we should be able to handle it. I know that by default System.Text.Json just returns null without calling converters when it encounters a `null` during deserialization, but it looks like we can override [`HandleNull`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonconverter-1.handlenull) in FSharp.STJ's converters to change this.

**Comment by cmeeren** [comment]
Happy that this is fixed! When will it be released?

**Comment by cmeeren** [comment]
Hi again, just wondering if it would be possible to create a new release with this change. 🙂 Otherwise I may have to work around it in other ways, which seems unnecessary since this is already implemented here, just not released yet.

**Comment by Tarmil** [comment]
Published in v1.2.

**Comment by cmeeren** [comment]
Thanks a lot!
